lms/djangoapps/ccx/tests/test_ccx_modulestore.py

```diff
@@ -0,0 +1,116 @@
+"""
+Test the CCXModulestoreWrapper
+"""
+from collections import deque
+from ccx_keys.locator import CCXLocator, CCXBlockUsageLocator
+import datetime
+from itertools import izip_longest
+import pytz
+from student.tests.factories import (  # pylint: disable=import-error
+    AdminFactory,
+    CourseEnrollmentFactory,
+    UserFactory,
+)
+from xmodule.modulestore.tests.django_utils import (
+    ModuleStoreTestCase,
+    TEST_DATA_SPLIT_MODULESTORE)
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+
+from ..models import CustomCourseForEdX
+
+
+def flatten(seq):
+    """
+    For [[1, 2], [3, 4]] returns [1, 2, 3, 4].  Does not recurse.
+    """
+    return [x for sub in seq for x in sub]
+
+
+class TestCCXModulestoreWrapper(ModuleStoreTestCase):
+    """tests for a modulestore wrapped by CCXModulestoreWrapper
+    """
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    def setUp(self):
+        """
+        Set up tests
+        """
+        super(TestCCXModulestoreWrapper, self).setUp()
+        course = CourseFactory.create()
+
+        # Create instructor account
+        coach = AdminFactory.create()
+
+        # Create a course outline
+        self.mooc_start = start = datetime.datetime(
+            2010, 5, 12, 2, 42, tzinfo=pytz.UTC)
+        self.mooc_due = due = datetime.datetime(
+            2010, 7, 7, 0, 0, tzinfo=pytz.UTC)
+        chapters = [ItemFactory.create(start=start, parent=course)
+                    for _ in xrange(2)]
+        sequentials = flatten([
+            [
+                ItemFactory.create(parent=chapter) for _ in xrange(2)
+            ] for chapter in chapters
+        ])
```

Comment:
You can actually just write this as:

    [ItemFactory.create(parent=chapter) for _ in xrange(2) for chapter in chapters]